src/FSharp.Core/local.fs

```diff
@@ -14,6 +14,11 @@ module internal DetailedExceptions =
         let msg = String.Format (format, paramArray)
         raise (new ArgumentException (msg, arg))
 
+    /// takes an argument, a formatting string, a param array to splice into the formatting string
+    let inline invalidArgOutOfRangeFmt (arg:string) (format:string) paramArray =
+        let msg = String.Format (format, paramArray)
```

Comment:
With #16556 in place, would it not be better to just use an interpolated string here?

Reply:
I'm not sure, this is an exceptional case, performance shouldn't matter here, so I think consistency with other cases is a bit more important. All cases could be refactored later all together.